cdc/sink/producer/kafka/kafka.go

```diff
@@ -345,7 +345,7 @@ func kafkaClientID(role, captureAddr, changefeedID, configuredClientID string) (
 	return
 }
 
-func validateMaxMessageBytesAndCreateTopic(admin kafka.ClusterAdminClient, topic string, config *Config) error {
+func validateMaxMessageBytesAndCreateTopic(admin kafka.ClusterAdminClient, topic string, config *Config, saramaConfig *sarama.Config) error {
```

Comment:
This func seems strange at the moment. Can we split it up?
1. getMaxMessageBytes is responsible for getting the right value
2. createTopic is used to create the topic

This way we won't have to pass two configurations and modify them at the same time. Now this function has become very complicated. Originally we didn't modify Sarama's configuration in this method. Now it is not only responsible for creating the topic, but also for setting the sarma configuration correctly, but this configuration is not directly related to creating the topic. It does too many things. I prefer to keep them separate.

Reply:
If there is a better way to get it not to set it up at the same time that would work too. I can only think of separating it at the moment.
The sarma configuration has been patched once above via kafka's configuration, but we're modifying it at the same time in this function.

It is currently confusing from either the caller's or the test code's point of view.

Reply:
This worthy another PR to do it.

Reply:
I think it might be better to do it in this time. It should just be a simple split method will work.

Reply:
I would leave that to another PR, as a future refinement.

Reply:
I still think it would be better to do this after the release, to prevent some potential new problems.